plugins/module_utils/nutanix_api_client.py

```diff
@@ -137,10 +138,44 @@ def delete_vm(vm_uuid, client):
     return response.json()["status"]["execution_context"]["task_uuid"]
 
 
-def list_images(filter, client):
-    image_list_response = client.request(
-        api_endpoint="v3/images/list", method="POST", data=json.dumps(filter))
-    return image_list_response.json()
+def update_powerstate_vm(vm_uuid, client, mechanism, power_state):
+
+    data = get_vm(vm_uuid, client)
+
+    if "status" in data:
+        del data["status"]
+
+    data["spec"]["resources"]["power_state"] = power_state
+
+    if "power_state_mechanism" in data["spec"]["resources"]:
+        data["spec"]["resources"]["power_state_mechanism"]["mechanism"] = mechanism
+    else:
+        data["spec"]["resources"] = {
+            "power_state_mechanism": {
+                "mechanism": mechanism
+            }
+        }
+
+    return update_vm(vm_uuid, data, client)
+
+
+def get_image_uuid(image_name, client):
+    length = 250
```

Comment:
Can we use global vars to set default value 
e.g. length = DEFAULT_LENGTH

Define global vars in some other file and import it  or in same file is also fine.